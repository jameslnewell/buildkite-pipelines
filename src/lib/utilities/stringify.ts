import yaml from 'js-yaml';
import prettier from 'prettier';
import {PipelineSchema} from '../schema';
import {PipelineBuilder} from '../builders';
import {isPipelineBuilder} from '../builders/isPipelineBuilder';

export async function stringify(
  pipeline: PipelineSchema | PipelineBuilder,
): Promise<string> {
  return prettier.format(
    yaml.dump(isPipelineBuilder(pipeline) ? await pipeline.build() : pipeline, {
      styles: {
        sortKeys: true,
        noRefs: true,
      },
    }),
    {
      parser: 'yaml',
    },
  );
}
