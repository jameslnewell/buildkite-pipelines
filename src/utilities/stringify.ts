import yaml from 'js-yaml';
import prettier from 'prettier';
import {PipelineSchema} from '../schema';

export function stringify(pipeline: PipelineSchema): string {
  return prettier.format(
    yaml.dump(pipeline, {
      styles: {
        sortKeys: true,
        noRefs: true,
      },
    }),
    {
      parser: 'yaml',
    },
  );
}
