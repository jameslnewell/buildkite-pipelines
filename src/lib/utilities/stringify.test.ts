import {stringify} from './stringify';

describe('stringify()', () => {
  test('outputs a yaml string', async () => {
    const result = await stringify({
      steps: [
        {
          label: 'lint',
          command: 'yarn run lint',
        },
      ],
    });
    expect(result).toMatchSnapshot();
  });
});
