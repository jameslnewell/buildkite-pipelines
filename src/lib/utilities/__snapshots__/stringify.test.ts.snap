// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`stringify() outputs a yaml string 1`] = `
"steps:
  - label: lint
    command: yarn run lint
"
`;
